File: 1094/solution.py
from typing import List


class Solution:
    def carPooling(self, trips: List[List[int]], capacity: int) -> bool:
        arr = [0] * 1002
        m = 0
        for n, f, t in trips:
            arr[f + 1] += n
            arr[t + 1] -= n
            m = max(m, t + 1)
        for i in range(1, m + 1):
            arr[i] += arr[i - 1]
            if arr[i] > capacity:
                return False
        return True

File: 1094/Solution.java
class Solution {
    public boolean carPooling(int[][] trips, int capacity) {
        int[] path = new int[1001];
        for (int[] trip: trips) {
            int num = trip[0], from = trip[1], to = trip[2];
            path[from] += num;
            path[to] -= num;
        }
        int total = 0;
        for (int i = 0; i <= 1000; i++) {
            total += path[i];
            if (total > capacity) {
                return false;
            }
        }
        return total <= capacity;
    }
}